src/clusterfuzz/_internal/metrics/monitor.py

```diff
@@ -111,41 +145,14 @@ def __init__(self):
 
   def run(self):
     """Run the flusher thread."""
-    project_path = _monitoring_v3_client.common_project_path(  # pylint: disable=no-member
-        utils.get_application_id())
-
+    should_stop = False
     while True:
-      try:
-        if self.stop_event.wait(FLUSH_INTERVAL_SECONDS):
-          return
-
-        time_series = []
-        end_time = time.time()
-        for metric, labels, start_time, value in _metrics_store.iter_values():
-          if (metric.metric_kind == metric_pb2.MetricDescriptor.MetricKind.GAUGE  # pylint: disable=no-member
-             ):
-            start_time = end_time
-
-          series = _TimeSeries()
-          metric.monitoring_v3_time_series(series, labels, start_time, end_time,
-                                           value)
-          time_series.append(series)
-
-          if len(time_series) == MAX_TIME_SERIES_PER_CALL:
-            time_series.sort(key=_time_series_sort_key)
-            _create_time_series(project_path, time_series)
-            time_series = []
-
-        if time_series:
-          time_series.sort(key=_time_series_sort_key)
-          _create_time_series(project_path, time_series)
-      except Exception as e:
-        if environment.is_android():
-          # FIXME: This exception is extremely common on Android. We are already
-          # aware of the problem, don't make more noise about it.
-          logs.warning(f'Failed to flush metrics: {e}')
-        else:
-          logs.error(f'Failed to flush metrics: {e}')
+      # Make sure there are no metrics left to flush after monitor.stop()
+      if self.stop_event.wait(FLUSH_INTERVAL_SECONDS):
+        should_stop = True
+      flush_metrics()
```

Comment:
Can you add details of this change to PR description also? Is this addressing a problem with the final set of metrics not being flushed?